src/test/java/de/unijena/cheminf/fragment/fingerprint/FragmentFingerprinterTest.java

```diff
@@ -0,0 +1,517 @@
+        File tmpResultsLogFile = new File(tmpWorkingPath + "/Fingerprints/" + FragmentFingerprinterTest.FINGERPRINTS_FILE_NAME  + ".txt");
+        FileWriter tmpResultsLogFileWriter = new FileWriter(tmpResultsLogFile, false);
+        PrintWriter tmpResultPrintWriter = new PrintWriter(tmpResultsLogFileWriter);
+        int[] tmpBitArray;
+        for (int tmpCurrentLine = 1; tmpCurrentLine < tmpList.size(); tmpCurrentLine++) {
+            tmpSeparateList = tmpList.get(tmpCurrentLine);
+            List<String> ListWithoutNameAndMoleculeSmiles = tmpSeparateList.subList(2, tmpSeparateList.size());
+            HashMap<String, Integer> tmpMoleculeFragmentsMap = new HashMap<>();
+            FragmentFingerprinterTest.dataForGenerationBitFingerprint = new ArrayList<>(FragmentFingerprinterTest.initialCapacityValue);
+            for (int i = 0; i < ListWithoutNameAndMoleculeSmiles.size(); i++) {
+                if (i % 2 == 0) {
+                    tmpMoleculeFragmentsMap.put(ListWithoutNameAndMoleculeSmiles.get(i), Integer.valueOf(ListWithoutNameAndMoleculeSmiles.get(i + 1)));
+                    FragmentFingerprinterTest.dataForGenerationBitFingerprint.add(ListWithoutNameAndMoleculeSmiles.get(i));
+                }
+            }
+            // Illustration the results of the bit arrays for the specified molecules
+            try {
+                IBitFingerprint tmpBitFingerprint = tmpFingerprintRepresentation.getBitFingerprint(FragmentFingerprinterTest.dataForGenerationBitFingerprint);
+                ICountFingerprint tmpCountFingerprint = tmpFingerprintRepresentation.getCountFingerprint(tmpMoleculeFragmentsMap);
+                tmpBitArray = tmpFingerprintRepresentation.getBitArray(FragmentFingerprinterTest.dataForGenerationBitFingerprint);
+                int tmpLength = java.util.Arrays.toString(tmpBitArray).length();
+                tmpCountFingerprint.setBehaveAsBitFingerprint(false);
+                System.out.println("\t\tNumber of positive bits " + tmpSeparateList.get(0) + ": " + tmpBitFingerprint.cardinality());
+                System.out.println("\t\tIndices of positive bits " + tmpSeparateList.get(0) + ": " + tmpBitFingerprint.asBitSet().toString());
+                System.out.println("\t\tCount for the bin with the index 5 " + tmpSeparateList.get(0) + ": " + tmpCountFingerprint.getCount(5));
+                tmpResultPrintWriter.println(java.util.Arrays.toString(tmpBitArray).substring(1, tmpLength - 1));
+                tmpResultPrintWriter.flush();
+            } catch( IllegalArgumentException  anException) {
+                throw anException;
+            } catch (NullPointerException anException) {
+                System.out.println("Generation of one fingerprint did not work.");
+            }
+            // add all molecule maps
+            FragmentFingerprinterTest.moleculeFragmentList.add(tmpMoleculeFragmentsMap);
+        }
+        // Objects necessary for the test are created (used only in @Test)
```

Comment:
One could say that only the things that follow now (and what is required for them) is the real setup, nothing more, referring to the other comment.